playbooks/openstack/e2e.yaml

```diff
@@ -0,0 +1,197 @@
+        k8s_version: "{{ (metadata_content['content'] | b64decode | from_yaml)['versions']['kubernetes'] }}"
+      when: metadata_content is defined
+    - name: Extract cloud name from clouds_yaml_full
+      ansible.builtin.set_fact:
+        cloud_name: "{{ clouds_yaml_full.clouds.keys() | first }}"
+      when: clouds_yaml_full.clouds is defined and clouds_yaml_full.clouds | dict2items | length == 1
+    - name: Generate clusterstack YAML
+      ansible.builtin.template:
+        src: "cluster-stack-template.yaml.j2"
+        dest: "{{ ansible_user_dir }}/clusterstack.yaml"
+        mode: "0644"
+    - name: Apply cluster-stack template
+      ansible.builtin.command: "kubectl apply -f {{ ansible_user_dir }}/clusterstack.yaml"
+      changed_when: true
+    - name: Pause for 2 minutes
+      ansible.builtin.pause:
+        minutes: 2
+    - name: Wait for cluster-stack to be ready
+      ansible.builtin.command: "kubectl wait clusterstack/clusterstack --for=condition=Ready --timeout=1440s"
+      changed_when: true
+    - name: Create k8s workload cluster and execute checks
+      block:
+        - name: Create Server Groups for nodes when scs_compliance tests are enabled
+          ansible.builtin.import_tasks: ../tasks/create_server_groups.yaml
+          when: scs_compliance.enabled
+        - name: Generate cluster YAML
+          ansible.builtin.template:
+            src: "cluster.yaml.j2"
+            dest: "{{ ansible_user_dir }}/cluster.yaml"
+            mode: "0644"
+          vars:
+            worker_server_group_id: "{{ srvgrp_worker.stdout | default('') }}"
+            worker_server_group_id_value: "{% if worker_server_group_id == '' %}\"\"{% else %}{{ worker_server_group_id }}{% endif %}"
+            controller_server_group_id: "{{ srvgrp_controller.stdout | default('') }}"
+            controller_server_group_id_value: "{% if controller_server_group_id == '' %}\"\"{% else %}{{ controller_server_group_id }}{% endif %}"
+        - name: Apply cluster template
+          ansible.builtin.command: "kubectl apply -f {{ ansible_user_dir }}/cluster.yaml"
+          changed_when: true
+        - name: Get kubeadmcontrolplane name
+          ansible.builtin.command: "kubectl get kubeadmcontrolplane -o=jsonpath='{.items[0].metadata.name}'"
+          register: kcp_name
+          changed_when: true
+        - name: Get kubeadmcontrolplane status
+          ansible.builtin.command: "kubectl wait kubeadmcontrolplane/{{ kcp_name.stdout }} --for=condition=Available --timeout={{ wait_for_cluster }}s"
+          changed_when: true
+        - name: Get kubeconfig of the workload k8s cluster
+          ansible.builtin.shell: "clusterctl get kubeconfig test-cluster > {{ cluster_manifest_dir }}/kubeconfig-test-cluster"
+          changed_when: true
+        - name: Wait for all system pods in the workload k8s cluster to become ready
```

Comment:
IMO, we should check also status of clusteraddon here

Reply:
AFAIK, all clusteraddon pods are deployed in the `kube-system` namespace, maybe except OCCM. I will check it and if so I can add a check for it.